src/Ionide.ProjInfo/Library.fs

```diff
@@ -630,12 +633,22 @@ module ProjectLoader =
 
             let pi = project.CreateProjectInstance()
 
-            let build = pi.Build(designTimeBuildTargets isLegacyFrameworkProjFile, loggers)
+            let doDesignTimeBuild () =
+                let build = pi.Build(designTimeBuildTargets isLegacyFrameworkProjFile, loggers)
+
+                if build then
+                    Ok(StandardProject pi)
+                else
+                    Error(sw.ToString())
+
+            let yieldTraversalProject () = Ok(TraversalProject pi)
+
+            // do traversal project detection here
+            match pi.GetProperty "IsTraversal" with
+            | null -> doDesignTimeBuild ()
+            | p when Boolean.Parse(p.EvaluatedValue) = false -> doDesignTimeBuild ()
+            | _ -> yieldTraversalProject ()
```

Comment:
The core change for the 'classic' workspace loader is here - depending on what kind of project this is we pivot behavior to just finding projectreferences, or doing a full design-time-build.